extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -68,97 +72,291 @@ export async function getRepositories(): Promise<string[] | undefined> {
-  let ref: string | undefined;
   let language: string | undefined;
-  let repositories: string[] | undefined;
+  if (await fs.pathExists(path.join(originalPackRoot, 'qlpack.yml'))) {
+    // don't include ql files. We only want the queryFile to be copied.
+    const toCopy = await cliServer.packPacklist(originalPackRoot, false);
+
+    // also copy the lock file (either new name or old name) and the query file itself. These are not included in the packlist.
+    [path.join(originalPackRoot, 'qlpack.lock.yml'), path.join(originalPackRoot, 'codeql-pack.lock.yml'), queryFile]
+      .forEach(absolutePath => {
+        if (absolutePath) {
+          toCopy.push(absolutePath);
+        }
+      });
+
+    let copiedCount = 0;
+    await fs.copy(originalPackRoot, queryPackDir, {
+      filter: (file: string) =>
+        // copy file if it is in the packlist, or it is a parent directory of a file in the packlist
+        !!toCopy.find(f => {
+          // Normalized paths ensure that Windows drive letters are capitalized consistently.
+          const normalizedPath = Uri.file(f).fsPath;
+          const matches = normalizedPath === file || normalizedPath.startsWith(file + path.sep);
+          if (matches) {
+            copiedCount++;
+          }
+          return matches;
+        })
+    });
 
-  // If the user has an explicit `.repositories` file, use that.
-  // Otherwise, prompt user to select repositories from the `codeQL.remoteQueries.repositoryLists` setting.
-  if (await fs.pathExists(repositoriesFile)) {
-    void logger.log(`Found '${repositoriesFile}'. Using information from that file to run ${queryFile}.`);
+    void logger.log(`Copied ${copiedCount} files to ${queryPackDir}`);
 
-    const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
+    language = await findLanguage(cliServer, Uri.file(targetQueryFileName));
 
-    ref = config.ref || 'main';
-    language = config.language || await findLanguage(cliServer, uri);
-    repositories = config.repositories;
   } else {
-    ref = 'main';
-    [language, repositories] = await Promise.all([findLanguage(cliServer, uri), getRepositories()]);
-  }
+    // open popup to ask for language if not already hardcoded
+    language = fallbackLanguage || await askForLanguage(cliServer);
 
+    // copy only the query file to the query pack directory
+    // and generate a synthetic query pack
+    // TODO this has a limitation that query packs inside of a workspace will not resolve it's peer dependencies.
+    // Something to work on later. For now, we will only support query packs that are not in a workspace.
+    void logger.log(`Copying ${queryFile} to ${queryPackDir}`);
+    await fs.copy(queryFile, targetQueryFileName);
+    void logger.log('Generating synthetic query pack');
+    const syntheticQueryPack = {
+      name: 'codeql-remote/query',
+      version: '1.0.0',
+      dependencies: {
+        [`codeql/${language}-all`]: '*',
```

Comment:
This part interests me. I take it this is the equivalent of https://github.com/dsp-testing/qc-run2/blob/ff207ffe5d100a9eb55743bc01f032336abdcde7/src/codeql.ts#L46? How does this behave in the following cases:

1. I've made a qlpack that doesn't depend on the core libraries (perhaps it is language independent). Odd case, I know.
2. I have a qlpack that contains just a query and the qlpack yml file. I've modified one of the core libraries, for example to add a new sink.
3. I have a qlpack that adds some new library files and also uses the core libraries.

Reply:
To be clear, this situation only exists when there is no qlpack associated with this query. So...

1. As long as there is a qlpack.yml that can be resolved and correctly declares itself to be no-lang, it should be fine.
2. I don't think we can (or should?) support this. The `codeql pack create` command compiles the current pack and also grabs all of its dependencies from the package cache, which get copied into the pack. Users are not supposed to manually edit these dependencies (it would be like manually editing your `node_modules`). If this is something that a user feels they need to do, then that's a missing feature we need to work on.
3. This is already handled by `codeql pack create`.

Reply:
> 2\. I don't think we can (or should?) support this. The `codeql pack create` command compiles the current pack and also grabs all of its dependencies from the package cache, which get copied into the pack. Users are not supposed to manually edit these dependencies (it would be like manually editing your `node_modules`). If this is something that a user feels they need to do, then that's a missing feature we need to work on.

That's fine for this PR but I think we are missing an important use-case where a CodeQL developer is working on a new or updated query that also requires changes to the core language libraries. I'll add it to the agenda for discussion.

Reply:
Hmmm...yes, something to discuss since I don't think we should be encouraging users to change the core libraries. Maybe there's a use case I'm missing.

But, even so, when we run the `pack create` command, all paths in the vscode workspace are added to the `--additional-packs` path, so if the core libraries are in the workspace, they _should_ be used instead of whatever is published in the package registry (I haven't tried this out yet, so not sure).